src/item.cpp

```diff
@@ -6582,6 +6582,13 @@ bool item::is_reloadable_helper( const itype_id &ammo, bool now ) const
                 return false;
             }
         }
+        //Now single shoted gun also has magazine_well slot for speedloader
+        //If ammo is not an ammo it may be dangerous to use parameters like ammo->ammo->type
+        //It is complicated: normal magazine in addition to speedloader? Magazines of mods?
+        if( !ammo->ammo ) {
```

Comment:
```suggestion
        if( now && !ammo->ammo ) {
```

Reply:
But seem like no real difference? It's deep in the night dont have clear headed now...

Reply:
Before, it would do:
```cpp
if( !now ) {
   return true;
} else {
   return ammo_remaining() < ammo_capacity( ammo->ammo->type );
}
```
Without this change, it does
```cpp
if( !now || !ammo->ammo ) {
   return magazine_compatible().count( ammo );
} else {
   return ammo_remaining() < ammo_capacity( ammo->ammo->type );
}
```
now it does:

```cpp
if( !now ) {
    return true;
else if( !ammo->ammo ) {
   return magazine_compatible().count( ammo );
} else {
   return ammo_remaining() < ammo_capacity( ammo->ammo->type );
}
```